From b627f31a437ff9e995eac14b13398fadacc7dd94 Mon Sep 17 00:00:00 2001
From: Feifei Wang <815101330@qq.com>
Date: Fri, 15 Aug 2025 09:25:53 +0800
Subject: [PATCH 13/13] fix support different queue depth bug

If user config different queue depth, driver will fix this.
---
 drivers/net/hinic3/hinic3_pmd_ethdev.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/net/hinic3/hinic3_pmd_ethdev.c b/drivers/net/hinic3/hinic3_pmd_ethdev.c
index 4c1e945..68fdf66 100644
--- a/drivers/net/hinic3/hinic3_pmd_ethdev.c
+++ b/drivers/net/hinic3/hinic3_pmd_ethdev.c
@@ -643,6 +643,13 @@ static int hinic3_rx_queue_setup(struct rte_eth_dev *dev, uint16_t qid,
 
 	nic_dev = HINIC3_ETH_DEV_TO_PRIVATE_NIC_DEV(dev);
 
+	/*Queue depth must be equal to queue 0*/
+	if (qid != 0 && (nb_desc != nic_dev->rxqs[0]->q_depth)) {
+		PMD_DRV_LOG(WARNING, "rxq%u depth:%u is not equal to queue0 depth:%u.\n",
+			qid, nb_desc, nic_dev->rxqs[0]->q_depth);
+		nb_desc = nic_dev->rxqs[0]->q_depth;
+	}
+
 	/* Queue depth must be power of 2, otherwise will be aligned up */
 	rq_depth = (nb_desc & (nb_desc - 1)) ?
 		((u16)(1U << (ilog2(nb_desc) + 1))) : nb_desc;
@@ -855,6 +862,13 @@ static int hinic3_tx_queue_setup(struct rte_eth_dev *dev, uint16_t qid,
 	nic_dev = HINIC3_ETH_DEV_TO_PRIVATE_NIC_DEV(dev);
 	hwdev = nic_dev->hwdev;
 
+	/*Queue depth must be equal to queue 0*/
+	if (qid != 0 && (nb_desc != nic_dev->txqs[0]->q_depth)) {
+		PMD_DRV_LOG(WARNING, "txq%u depth:%u is not equal to queue0 depth:%u.\n",
+			qid, nb_desc, nic_dev->txqs[0]->q_depth);
+		nb_desc = nic_dev->txqs[0]->q_depth;
+	}
+
 	/* Queue depth must be power of 2, otherwise will be aligned up */
 	sq_depth = (nb_desc & (nb_desc - 1)) ?
 		   ((u16)(1U << (ilog2(nb_desc) + 1))) : nb_desc;
-- 
2.45.1.windows.1

